### PR TITLE
feat: add uninstall command

### LINKED ISSUE #709: feat(dart_frog_cli): include `uninstall` command to uninstall `dart_frog_cli`
**Description**

As a user I would like to uninstall `dart_frog_cli` by using `dart_frog uninstall`, this should:

- Remove global files created when installing `dart_frog_cli` (for example, completion files)
- Deactivate `dart_frog_cli`

**Additional Context**

- Useful for sharing uninstallation logic between IDE extensions.
- Improve user's experience since currently a full uninstall requires:
```sh
dart_frog uninstall-completion-files # Hidden command most users are not aware of.
dart pub global deactivate dart_frog_cli
```

### PR DESCRIPTION
**Pull request by jsgalarraga**
## Status

**READY**

## Description

- Added uninstall section in the documentation.
- Added uninstall command that prints the link to the documentation.

Fixes #709 

Screenshot 2023-07-27 at 14 03 53

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: docs/docs/overview.md [hunk 1]
```diff
@@ -69,6 +69,22 @@ Create a production build which includes a `DockerFile` so that you can deploy a
...
+# 🗑️ Uninstall the dart_frog cli completion files
+dart_frog uninstall-completion-files
+```
+
+Now, dart frog can be uninstalled safely.
+
+```shell
+# 🗑️ Uninstall the dart_frog cli from pub.dev
+dart pub global deactivate dart_frog_cli
+```
```
Comment (renancaraujo) [inline docs/docs/overview.md:86]: Hello mr @jsgalarraga , can you split this bit into as different pr? The ci would make this to be included in site before we release the DF CLI with the changes below.

Comment (jsgalarraga) [reply docs/docs/overview.md:86]: Sure!

File: packages/dart_frog_cli/lib/src/commands/uninstall/uninstall.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  String get description => 'Explains how to uninstall the Dart Frog CLI.';
+
+  @override
+  String get name => 'uninstall';
+
+  @override
+  Future<int> run() async {
+    _logger.info(
+      'For instructions on how to uninstall $packageName completely, check out:'
+      '\nhttps://dartfrog.vgv.dev/docs/overview#uninstalling-',
```
Comment (alestiago) [inline packages/dart_frog_cli/lib/src/commands/uninstall/uninstall.dart:27]: We should use a link instead, see [this](https://github.com/VeryGoodOpenSource/dart_frog/blob/060c593baa50d9b6489e1cee5300a7277f80160c/packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart#L313) as an example.

File: packages/dart_frog_cli/test/src/commands/uninstall/uninstall_test.dart [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import 'package:dart_frog_cli/src/command_runner.dart';
+import 'package:dart_frog_cli/src/commands/uninstall/uninstall.dart';
+import 'package:mason/mason.dart' hide packageVersion;
+import 'package:mocktail/mocktail.dart';
+import 'package:test/test.dart';
+
+class MockLogger extends Mock implements Logger {}
```
Comment (alestiago) [inline packages/dart_frog_cli/test/src/commands/uninstall/uninstall_test.dart:7]: **Nit:**

```suggestion
class _MockLogger extends Mock implements Logger {}
```

**Review by renancaraujo — APPROVED** [review]

**Pull request merged**